### PR TITLE
add users endpoint

### PR DESCRIPTION
**Pull request by ewangler**
/api/users

### CODE REVIEW

File: backend/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,31 @@
 class ApplicationController < ActionController::API
+  rescue_from ActiveRecord::RecordNotFound do |e|
+    render_error({ message: e.message }, :not_found)
+  end
+
+  rescue_from ActiveRecord::Rollback do |e|
```
Comment (sihu) [inline backend/app/controllers/application_controller.rb]: according to documentation it should be possible to pass in multiple classes:

```rb
rescue_from ActiveRecord::Rollback, ActiveRecord::RecordInvalid, ActiveRecord::InvalidForeignKey, ActionController::ParameterMissing   do |e|
   render_error({ message: e.message }, :unprocessable_entity)

File: backend/config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
+
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
```
Comment (sihu) [inline backend/config/initializers/devise.rb]: i tried to comment this already: i would add `config.secret_key` and `config.pepper` (now commented out) as and `config.mailer_sender ` ENVs to this file. 
Plus i would delete the rest of the comments. or at least the description.

File: backend/db/migrate/20180930091501_add_devise_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, :reset_password_token, unique: true
+    # add_index :users, :confirmation_token,   unique: true
+    # add_index :users, :unlock_token,         unique: true
+  end
+
+  def self.down
+    # By default, we don't want to make any assumption about how to roll back a migration when your
```
Comment (sihu) [inline backend/db/migrate/20180930091501_add_devise_to_users.rb]: hmm..not sure whether we should do something about that.

File: backend/db/migrate/20180930091501_add_devise_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
```
Comment (sihu) [inline backend/db/migrate/20180930091501_add_devise_to_users.rb]: also here i would delete the unwanted fields

**Review by sihu — APPROVED** [review]
LGTM 👍 some small comments

File: backend/app/controllers/application_controller.rb (hunk 1 above)
Comment (ewangler) [reply backend/app/controllers/application_controller.rb]: ah nice! thanks

File: backend/config/initializers/devise.rb (hunk 2 above)
Comment (ewangler) [reply backend/config/initializers/devise.rb]: thanks, i will move this to a seperate PR and set up devise correctly

File: backend/app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,17 @@
 class ApplicationController < ActionController::API
+  rescue_from ActiveRecord::RecordNotFound, ActiveRecord::InvalidForeignKey,
+              ActiveRecord::RecordInvalid, ActiveRecord::Rollback,
+              ActionController::ParameterMissing do |e|
+    render_error({ message: e.message }, :not_found)
```
Comment (sihu) [inline backend/app/controllers/application_controller.rb]: sorry now i confused you: i think this was the only message rendering `:not_found` while all others could be merged and return `render_error({ message: e.message }, :unprocessable_entity)`

**Pull request merged**